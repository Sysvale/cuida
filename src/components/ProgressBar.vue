<template>
	<div
		class="progress-bar"
	>
		<span
			v-if="showText"
			:class="textAside ? 'bar__text--aside' : 'bar__text'"
		>
			{{ formatedvalue }}
		</span>

		<div
			class="progress-bar__content"
		>
			<div
				:class="`progress-bar__indicator--${variant}`"
				:style="progressIndicatorStyle"
			/>
		</div>
	</div>
</template>

<script>

export default {
	props: {
		/**
		 * A variante de cor. São 9 variantes implementadas: 'green', 'teal', 'turquoise',
		 * 'blue', 'indigo', 'violet', 'pink', 'red', 'orange' e 'amber'.
		 */
		variant: {
			type: String,
			default: 'green',
		},
		/**
		 * Define o indicador de progresso da ProgressBar.
		 */
		value: {
			type: Number,
			default: 0,
			required: true,
		},
		/**
		 * Define se a porcentagem vai ser mostrada.
		 */
		showText: {
			type: Boolean,
			default: false,
			required: false,
		},
		/**
		 * Quando true, dispõe o texto à direita.
		 */
		textAside: {
			type: Boolean,
			default: false,
		},
	},

	computed: {
		formatedvalue() {
			return `${parseInt(this.value * 100, 10)}%`;
		},

		progressIndicatorStyle() {
			return {
				'--width': this.formatedvalue,
				'--leftMargin': this.textAside ? '3' : '10',
				'--bottomMargin': this.textAside ? '10' : '2',
			};
		},

		textDirection() {
			return this.textAside ? 'row-reverse' : 'column';
		},
	},
};
</script>

<style lang="scss" scoped>
@use './../assets/sass/tokens/index' as tokens;
.progress-bar {
	align-items: center;
	display: flex;
	flex-direction: v-bind(textDirection);

	&__content {
		background-color: tokens.$n-40;
		border-radius: tokens.$border-radius-extra-large;
		height: 7px;
		width: 100%;
	}

	&__indicator {
		@include tokens.variantResolver using ($color-name, $shade-50, $shade-100, $shade-200, $shade-300, $base-color, $shade-500, $shade-600) {
			background-color: $base-color;
			border-radius: tokens.$border-radius-extra-large;
			height: 7px;
			width: var(--width);
		}
	}

	&__text--aside {
		font-weight: tokens.$font-weight-semibold;
		margin: ml(3);
	}

	&__text {
		font-weight: tokens.$font-weight-semibold;
		margin: mb(2);
	}
}
</style>
