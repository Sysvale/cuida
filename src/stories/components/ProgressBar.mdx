import { Meta, Story, Props, Controls, Canvas, Preview } from '@storybook/blocks';
import ProgressBar from '../../components/ProgressBar.vue';
import * as ProgressBarStories from './ProgressBar.stories';

<Meta of={ProgressBarStories} />

export const Template = (args) => ({
	components: { CdsProgressBar: ProgressBar },
	setup() {
		return { args };
	},
	template: /*html*/ `<cds-progress-bar v-bind="args"/>`,
});

# ProgressBar

### ProgressBars são componentes utilizados para indicar o status e progresso de algo aos usuários.

***

<br />

## Quando usar:

* For necessário transmitir a ideia do progresso de algo.
* For necessário transmitir a ideia do status de algo.
* Se deseja chamar a atenção do usuário para uma informação específica de maneira gráfica.

<br />

## Quando não usar:

* For necessário criar loaders.
* For necessária muita precisão ao mostrar os números ou progresso.
* For necessário criar steppers.

<br />

## Preview

<Canvas>
  <Story of={ProgressBarStories.ProgressBar} />
</Canvas>

<Controls />
