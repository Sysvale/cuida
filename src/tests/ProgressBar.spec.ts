import { describe, test, expect } from 'vitest';
import ProgressBar from '../components/ProgressBar.vue';
import { mount } from '@vue/test-utils';

describe('ProgressBar', () => {
	test('renders correctly', async () => {
		const wrapper = mount(ProgressBar, {
			props: {
				value: 0.67,
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
