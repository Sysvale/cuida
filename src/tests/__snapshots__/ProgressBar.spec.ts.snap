// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar > renders correctly 1`] = `
"<div data-v-84df1bb9="" class="progress-bar">
  <!--v-if-->
  <div data-v-84df1bb9="" class="progress-bar__content">
    <div data-v-84df1bb9="" class="progress-bar__indicator--green" style="--width: 67%; --leftMargin: 10; --bottomMargin: 2;"></div>
  </div>
</div>"
`;
